IPs/RSKIP502.md

```diff
@@ -0,0 +1,252 @@
+---
+rskip: 502
+title: PowPeg and Union Bridge integration
+created: 19-MAR-25
+author: MI
+purpose: Sca
+layer: Core
+complexity: 1
+status: Draft
+description: 
+---
+
+|RSKIP          |502           |
+| :------------ |:-------------|
+|**Title**      |PowPeg and Union Bridge integration |
+|**Created**    |19-MAR-25 |
+|**Author**     |MI |
+|**Purpose**    |Sca |
+|**Layer**      |Core |
+|**Complexity** |1 |
+|**Status**     |Draft |
+
+## Abstract
+
+This RSKIP proposes the integration of the Union Bridge [[1]](#references)  with the existing PowPeg to enable secure RBTC transfers between the two systems. The PowPeg will facilitate the transfer of RBTC to the Union Bridge, with the addition of a locking cap to mitigate risks by limiting the total amount of RBTC that can be transferred. The integration will extend the current Bridge contract's functionality, incorporating new methods for managing transfers, updating the Union Bridge contract address, and adjusting the locking cap. These changes will ensure that the PowPeg remains secure and robust while supporting the ongoing development of the Union Bridge. Special considerations have been made to allow seamless testing and updates during the Union Bridge's development phase, with restrictions placed on mainnet interactions to avoid unauthorized changes.
+
+## Motivation
+
+The Union Bridge is a new project currently under development that will interact with the Rootstock network. Integrating the Union Bridge with the existing PowPeg is essential to facilitate the transfer of RBTC. A key concern is ensuring that the integration remains secure, and that risks are minimized through a locking cap that controls the amount of RBTC transferred. The PowPeg's current functionality must be extended with minimal disruption, while also accounting for the need to manage the evolving nature of the Union Bridge.
+
+Additionally, the Union Bridge is still in development, and its contract address is likely to change as it evolves. To accommodate this, the proposed solution includes a flexible mechanism for updating the Union Bridge contract address during its testing and development phases, ensuring that these changes can occur securely without risking mainnet operations.
+
+This proposal aims to ensure that the PowPeg remains secure while enabling a smooth transition and integration with the Union Bridge as it matures, all while minimizing risks and maintaining full compatibility with the current system.
+
+## Specification
+
+### 1. New storage entry for Union Bridge contract address
+
+A new storage entry is necessary to store the address of the Union Bridge contract. This will allow the Bridge to identify when transactions are coming from the Union Bridge, and provide a mechanism for managing these interactions securely.
+
+Storage entry name: `unionBridgeContractAddress`
+
+Maximum size: 32 bytes
```

Comment:
Why is it 32 bytes? Should this be 20 bytes?

Reply:
Agree. Good catch.

Reply:
Good catch, thanks